### PR TITLE
deprecated forward()

### PR DESCRIPTION
**Pull request by jeabakker**

### CODE REVIEW

File: docs/contribute/code.rst [hunk 1]
```diff
@@ -444,8 +444,7 @@ developers looking at the code should be discouraged from refactoring in a way t
 
     // Can't use empty()/boolean: "0" is a valid value
     if ($str === '') {
-        register_error(elgg_echo('foo:string_cannot_be_empty'));
-        forward(REFERER);
+    	throw new \Elgg\Exceptions\Http\BadRequestException(elgg_echo('foo:string_cannot_be_empty'));
```
Comment (jdalsem) [inline docs/contribute/code.rst]: incorrect indentation?

File: docs/design/database.rst [hunk 2]
```diff
@@ -214,25 +214,26 @@ your new element (defaulting to the current user's container):
...
     $user = elgg_get_logged_in_user_entity();
-    $container_guid = (int)get_input('container_guid');
+    $container_guid = (int) get_input('container_guid');
     
     if ($container_guid) {
     	$container = get_entity($container_guid);
     	
-        if (!$container->canWriteToContainer($user->guid)) {
+        if (!$container instanceof \ElggEntity || !$container->canWriteToContainer($user->guid)) {
             // register error and forward
```
Comment (jdalsem) [inline docs/design/database.rst]: remove comment

File: docs/design/database.rst [hunk 3]
```diff
@@ -214,25 +214,26 @@ your new element (defaulting to the current user's container):
...
 
-    $object = new ElggObject;
+    $object = new ElggObject();
     $object->container_guid = $container_guid;
 
     ...
 
-    $container = get_entity($container_guid);
-    forward($container->getURL());
+	// redirect to the created object
```
Comment (jdalsem) [inline docs/design/database.rst]: indentation?

**Review by jdalsem — APPROVED** [review]

File: docs/design/database.rst (hunk 2 above)
Comment (jeabakker) [reply docs/design/database.rst]: done

File: docs/design/database.rst (hunk 3 above)
Comment (jeabakker) [reply docs/design/database.rst]: fixed

File: docs/contribute/code.rst (hunk 1 above)
Comment (jeabakker) [reply docs/contribute/code.rst]: fixed

**Pull request merged**